tests/tests/test_compat.py

```diff
@@ -307,16 +311,46 @@ def compatibility_test_impl(self, env):
             assert_successful_deployment(api_deployments, deployment_id)
 
 
-@pytest.mark.skip(reason="See QA-443")
 class TestClientCompatibilityOpenSource(TestClientCompatibilityBase):
     def test_compatibility(self, setup_os_compat):
-        self.compatibility_test_impl(setup_os_compat)
+        for version in [
+            "2.0",
+            "2.1",
+            "2.2",
+            "2.3",
+            "2.4",
+            "2.5",
+            "2.6",
+            "3.0",
+            "3.1",
+            "mender-3.2",
+            "mender-3.3",
+            "mender-3.4",
+        ]:
```

Comment:
This list we will need to update after every release. Therefore:
* Set it in a single place, with a meaningful name like `COMPAT_MENDER_VERSIONS`
* Add a comment with instructions on how to update, like "After a new release of the client components, add the corresponding `extra/integration-testing/docker-compose.compat-X-X.yml` and update this list" ...

Reply:
fixed. thank you.

Reply:
you could also get the versions file and generate the list of versions from there...

Reply:
wait, sorry, what versions file?

Reply:
I think he meant the composition files, which is a nice maintainability improvement.

Reply:
I do not think so :) "versions file" != "composition files" what I did now is: scan the directory for the versions.

Reply:
yeah I meant the json file, and then generate the composition file names or even files from there - they should surely follow a naming pattern and we would have to rely on a single place to update instead of adding more items to the (post) release checklist...

Reply:
but yeah scanning the directory is also a good option!

Reply:
> yeah I meant the json file

One advantage of using the json file would be than after a new release, even the older servers (iow release branches from integration repo) would run the test with newer clients :rocket: However as we move out to rolling releases for the backend, then this specific advantage will lose value.

In any case the reason I thought you didn't mean this is because Piotr already tried to generate the compose files based on a template and it didn't work out. Due to the certificate changes and other minor things we have done along the way, different images need different tweaks to get them running.

Scanning the directory is a good improvement regardless. Well done @merlin-northern